utils/swift_build_support/swift_build_support/products/swiftinspect.py

```diff
@@ -77,13 +77,29 @@ def get_dependencies(cls):
                 swiftpm.SwiftPM]
 
 
-def run_build_script_helper(host_target, product, args):
-    toolchain_path = args.install_destdir
+def _get_toolchain_path(host_target, product, args):
+    # this logic initially was inside run_build_script_helper
+    # and was factored out so it can be used in testing as well
+
+    install_destdir = args.install_destdir
+    if swiftpm.SwiftPM.has_cross_compile_hosts(args):
+        install_destdir = swiftpm.SwiftPM.get_install_destdir(args,
+                                                              host_target,
+                                                              product.build_dir)
+    toolchain_path = targets.toolchain_path(install_destdir,
+                                            args.install_prefix)
```

Comment:
Try using [this better two-line formulation I came up with for other build products](https://github.com/apple/swift/pull/36917/files#diff-57b5bec593d589a8abf555dcffcdd0748b1dc27cbc7beec0b0891c41fd274909R90), though as above, you will have to do something about that Darwin prefix added below too.